MaxText/tests/max_utils_test.py

```diff
@@ -97,6 +101,108 @@ def test_init_training_state(self):
         max_utils.calculate_num_params_from_pytree(state.params), max_utils.calculate_num_params_from_pytree(self.params)
     )
 
+  def test_get_abstract_state_training(self):
```

Comment:
Before merging this PR, lets please talk about the impact of these tests on our CI time, and how to deal with that. 
CC: @richjames0 @gobbleturk